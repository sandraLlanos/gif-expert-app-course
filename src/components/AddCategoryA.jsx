import React from 'react';
import { useState } from "react";
import PropTypes from 'prop-types'

// eslint-disable-next-line react/prop-types
export const AddCategoryA = ({onNewCategory}) => {
    const [inputValue, setInputValue] = useState('')
    const onInputChange = (event) => {
        setInputValue(event.target.value)
    }
    const onSubmit = (event) => {
        event.preventDefault();
        if(inputValue.trim().length <= 1) return;
        onNewCategory( inputValue.trim() )
        setInputValue('');
    }

    return (
        <form onSubmit={onSubmit} aria-label='form'>
          <input 
            type="text"
            placeholder="Search"
            value={inputValue}
            onChange={(event) => onInputChange(event)}
          />
        </form>
        )
}


AddCategoryA.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
}

