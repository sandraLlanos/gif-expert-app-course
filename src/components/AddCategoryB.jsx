import { useState } from "react";

export const AddCategory = ({ onAddCategories }) => {
  // console.log('cat',onAddCategories)

  const [inputValue, setInputValue] = useState('One Push')

  const onInputChange = (event) => {
    // console.log('event :>> ', event.target.value);
    setInputValue(event.target.value)
  }

  const onSubmit = (event) => {
    event.preventDefault();
    // console.log(inputValue);
    // the input value will update the parent
    // communication btw components through props
    // onAddCategories(categories => {
    //   console.log([...categories, inputValue])
    // })
    if (inputValue.trim().length <=1) return;
    onAddCategories(categories => [...categories, inputValue]);
    // Clean Input value
    setInputValue('');
  }

  // you can use Form as a
  //  parent as fragments do, you can't use sibling elements without a wrapper
  return (
    <form onSubmit={onSubmit}>
      <input 
        type="text"
        placeholder="Search"
        value={inputValue}
        onChange={(event) => onInputChange(event)}
      />
    </form>
  )
}

