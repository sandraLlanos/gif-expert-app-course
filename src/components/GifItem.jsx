// eslint-disable-next-line react/prop-types
export const GifItem = ({title, url}) => {
    // console.log('title :>> ', title);
    // console.log('url :>> ', url);
    
  return (
    <div className="card">
        <img src={url} alt={title} />
        <p>{title}</p>
    </div>
  )
}
