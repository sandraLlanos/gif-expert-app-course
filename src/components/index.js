export * from './AddCategoryA';
export * from './AddCategoryB';
export * from './GifGrid';
export * from './GifItem';