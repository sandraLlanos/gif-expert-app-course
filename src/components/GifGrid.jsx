// import { useEffect, useState } from "react";
import React from 'react';
import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";
import PropTypes from 'prop-types';
// import { getGifs } from "../helpers/getGifs";

/* eslint-disable react/prop-types */
export const GifGrid = ({category}) => {

    const {images, isLoading } = useFetchGifs( category );
    // console.log({images, isLoading});
    
    
    return (
        <>
            <h3>{category}</h3>
            {
                isLoading &&  (<h2>Loading...</h2>)
            }           

            <div className="card-grid">
                {
                    images.map((image)=>(
                    //   <li key={id}>{title}</li>
                        <GifItem 
                            key={image.id}
                            { ...image }
                        />
                    ))
                }
            </div>
        </>
    )
}

GifGrid.prototypes = {
    category: PropTypes.string.isRequired,
}
