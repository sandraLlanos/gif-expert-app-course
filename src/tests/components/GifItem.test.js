import React from 'react';
import { render, screen } from "@testing-library/react"
import { GifItem } from "../../components/GifItem"

describe('test in GifItem', () => { 
    const title = 'Saitama';
    const url = 'https://one-punch.com/saitama.jpg';

    test('It should match with a snapshot', () => {
        const {container} = render( <GifItem title={title} url={url}/>);
        expect( container ).toMatchSnapshot();
    });

    test('It should show the image with the correct url and the ALT text', () => { 
        render( <GifItem title={title} url={url}/>);
        screen.debug();
        // expect(screen.getByRole( 'img' ).src).toBe( url );
        // expect(screen.getByRole( 'img' ).alt).toBe( title );
        const {src, alt} = screen.getByRole('img');
        expect( src ).toBe( url);
        expect( alt ).toBe( alt );
    });

    test('It should show the title in the component', () => { 
        render( <GifItem title={title} url={url}/>);
        expect( screen.getByText(title) ).toBeTruthy();
    })
})