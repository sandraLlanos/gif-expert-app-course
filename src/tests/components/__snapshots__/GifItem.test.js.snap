// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in GifItem It should match with a snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
